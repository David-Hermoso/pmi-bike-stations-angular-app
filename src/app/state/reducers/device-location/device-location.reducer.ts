import {createReducer, on} from '@ngrx/store';
import * as DeviceLocationActions from '../../actions/device-location.actions';
import {AppState} from '../../../models/app.state';

export const selectDeviceLocation = (appState: AppState) => appState.deviceLocation;

export const deviceLocationReducer = createReducer(
  undefined as unknown,
  on(DeviceLocationActions.deviceLocationSuccess,
    (currentState: any, {deviceLocation}) => deviceLocation
  )
);
