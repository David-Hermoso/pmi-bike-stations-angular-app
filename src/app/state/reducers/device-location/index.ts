export * from './device-location.reducer';
