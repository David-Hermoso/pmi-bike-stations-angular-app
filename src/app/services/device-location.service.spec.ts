import {DeviceLocationService} from './device-location.service';
import {of} from 'rxjs';

describe('DeviceLocationService', () => {
  it('should call getCurrentPosition', () => {
    const mockStore = {
      dispatch: jest.fn(),
      pipe: jest.fn(() => of([
        {
          img: '003.jpeg',
          id: '01',
          name: 'PLAÃ‡A DELS PATINS',
          lng: 2.659399509,
          lat: 39.56590061
        }
      ]))
    };
    const service: DeviceLocationService = new DeviceLocationService(mockStore as any);
    // @ts-ignore
    navigator.geolocation = {
      getCurrentPosition: jest.fn().mockImplementation((success) =>
        Promise.resolve(
          success({
            coords: {
              latitude: 10,
              longitude: 10
            }
          })
        )
      )
    };

    service.requestDeviceLocation();

    expect(navigator.geolocation.getCurrentPosition).toHaveBeenCalled();
  });
  it('should return distance between coordinates', () => {
    const from = {
      lat: 2.83,
      lng: 3.99
    };
    const to = {
      lat: 2.30,
      lng: 3.39
    };

    expect(DeviceLocationService.distanceInKmBetweenEarthCoordinates(from, to)).toMatchSnapshot();
  });
  it('should return empty string', () => {
    const from = undefined;
    const to = undefined;

    // @ts-ignore
    expect(DeviceLocationService.distanceInKmBetweenEarthCoordinates(from, to)).toStrictEqual('');
  });
});
