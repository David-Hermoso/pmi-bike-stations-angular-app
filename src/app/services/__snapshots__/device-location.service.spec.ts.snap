// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceLocationService should return distance between coordinates 1`] = `"88.97"`;
